src/RenderView.tsx

```diff
@@ -66,6 +74,59 @@ export default function RenderView(props: Props) {
 		return result;
 	});
 
+	const [cameraControls] = useState(() => new PointerLockControls(camera, renderer.domElement));
+
+	/* Use camera controls */
+	useEffect(() => {
+		const moveMeters = 0.05;
+		/* Only allow flight while mouse is in visualization area */
+		renderer.domElement.addEventListener('mouseenter', () => document.addEventListener("keydown", keyPressHandler, false), false);
+		renderer.domElement.addEventListener('mouseleave', () => document.removeEventListener("keydown", keyPressHandler, false), false);
+		/* Flight controls (keys) */
+		const keyPressHandler = (e: KeyboardEvent) => {
+			switch (e.key) {
```

Comment:
I think using `e.code` here would be better since it works with shift & non-qwerty layouts too

Reply:
Ah, good to know. Also done.